src/django_idom/templatetags/idom.py

```diff
@@ -34,17 +35,23 @@ def component(dotted_path: str, **kwargs: Any):
         </body>
         </html>
     """
-    _register_component(dotted_path)
-
+    component = _register_component(dotted_path)
+    uuid = uuid4().hex
     class_ = kwargs.pop("class", "")
-    json_kwargs = json.dumps(kwargs, separators=(",", ":"))
+
+    # Store the component's args/kwargs in the database if needed
+    # This will be fetched by the websocket consumer later
+    if func_has_params(component):
```

Comment:
I don't think we should quietly ignore parameters that may not be consumed by the component. It seems better to let the error happen and allow the user to correct it. I understand this is a performance optimization, but I think we can worry about that if/when it turns out to be necessary to make things more efficient.

Reply:
I [added in a check](https://github.com/idom-team/django-idom/pull/120/commits/e686894db9557ffbf8f30c0573b766f669f51c38) to catch this edge case.

I'd really like to avoid hitting the DB when possible, given how slow the Django ORM is.

Reply:
If that's the case, I'd recommend using `inspect.Signature` [to test](https://docs.python.org/3/library/inspect.html#inspect.Signature.bind) whether or not the `kwargs` will work with the given function:

```python
kwargs = {...}
func = lambda: ...
sig = inspect.signature(func)
# check if kwargs work with func
sig.bind(kwargs)
```

If you check this when rendering the template, you an safely avoid hitting the DB if the func has no args/kwargs at runtime.

Reply:
Implemented in https://github.com/idom-team/django-idom/pull/120/commits/4f4197513878cc8c8672d07579a8e71e6266628a